juriscraper/opinions/united_states/state/conn.py

```diff
@@ -2,47 +2,46 @@
 CourtID: conn
 Court Short Name: Conn.
 Author: Asadullah Baig <asadullahbeg@outlook.com>
-
 History:
- - 2014-07-11: created
- - 2014-08-08, mlr: updated to fix InsanityError on case_dates
- - 2014-09-18, mlr: updated XPath to fix InsanityError on docket_numbers
- - 2015-06-17, mlr: made it more lenient about date formatting
- - 2016-07-21, arderyp: fixed to handle altered site format
- - 2017-01-10, arderyp: restructured to handle new format use case that includes
+- 2014-07-11: created
+- 2014-08-08, mlr: updated to fix InsanityError on case_dates
+- 2014-09-18, mlr: updated XPath to fix InsanityError on docket_numbers
+- 2015-06-17, mlr: made it more lenient about date formatting
+- 2016-07-21, arderyp: fixed to handle altered site format
+- 2017-01-10, arderyp: restructured to handle new format use case that includes
         opinions without dates and flagged for 'future' publication
+- 2022-02-02, satsuki-chan: Fixed docket and name separator, changed super
+    class to OpinionSiteLinear
 """
 
 from datetime import date
 
 from lxml import etree
 
+from juriscraper.AbstractSite import logger
 from juriscraper.lib.string_utils import convert_date_string, normalize_dashes
-from juriscraper.OpinionSite import OpinionSite
+from juriscraper.OpinionSiteLinear import OpinionSiteLinear
 
 
-class Site(OpinionSite):
+class Site(OpinionSiteLinear):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.crawl_date = date.today()
-        self.url = "http://www.jud.ct.gov/external/supapp/archiveAROsup{year}.htm".format(
-            year=self.crawl_date.strftime("%y")
-        )
         self.court_id = self.__module__
+        current_year = date.today().strftime("%y")
+        self.url = f"http://www.jud.ct.gov/external/supapp/archiveAROsup{current_year}.htm"
+        self.status = "Published"
         self.cases = []
 
-    def _download(self, request_dict={}):
-        html = super()._download(request_dict)
-        self._extract_cases_from_html(html)
-        return html
+    def _process_html(self) -> None:
+        """Process the html and extract out the opinions
 
-    def _extract_cases_from_html(self, html):
-        """Build list of data dictionaries, one dictionary per case (table row)."""
+        :return: None
+        """
         # Strip inconsistently placed <font> and <br>
         # tags that make stable coverage almost impossible
-        etree.strip_tags(html, "font", "br")
+        etree.strip_tags(self.html, "font", "br")
         path = '//table[@id="AutoNumber1"]//ul'
-        for ul in html.xpath(path):
+        for ul in self.html.xpath(path):
```

Comment:
Instead of iterating over the Ups and then doing all that processing 

I think you could just do something like 

`for row in self.html.xpath("//td/font/ul/li | //td/ul/li"):`

I tested it and it handles all the edge cases.

Reply:
this also lets you grab the dates with something like this. 

```
date_raw = row.xpath(".//parent::ul/preceding-sibling::*")[-1].text_content()
dates = re.findall(r"\d{1,2}/\d{1,2}/\d{2,4}", date_raw)
```

you can avoid future dates now because dates will be empty or contain the date

Reply:
XPath changed to search dates and LI tags.